addons/account/models/account_move.py

```diff
@@ -4024,22 +4013,11 @@ def _add_lines_to_exchange_difference_vals(lines, exchange_diff_move_vals):
 
                 exchange_diff_move_vals['date'] = max(exchange_diff_move_vals['date'], line.date)
 
-                if not line.company_currency_id.is_zero(line.amount_residual):
+                if not line.company_currency_id.is_zero(line.amount_residual)\
+                        or line.currency_id and not line.currency_id.is_zero(line.amount_residual_currency):
                     # amount_residual_currency == 0 and amount_residual has to be fixed.
 
-                    if line.amount_residual > 0.0:
-                        exchange_line_account = journal.default_debit_account_id
-                    else:
-                        exchange_line_account = journal.default_credit_account_id
-
-                elif line.currency_id and not line.currency_id.is_zero(line.amount_residual_currency):
-                    # amount_residual == 0 and amount_residual_currency has to be fixed.
-
-                    if line.amount_residual_currency > 0.0:
-                        exchange_line_account = journal.default_debit_account_id
-                    else:
-                        exchange_line_account = journal.default_credit_account_id
-
+                    exchange_line_account = journal.default_account_id
```

Comment:
Restore this part. You need to retrieve `Exchange difference Loss` & `Exchange difference Gain` accounts from the company.